pkg/apis/alibabacloudprovider/v1beta1/alibabacloudmachineproviderconfig_types.go

```diff
@@ -139,13 +136,10 @@ type AlibabaCloudMachineProviderConfig struct {
 	//This parameter is empty by default.
 	Description string `json:"description,omitempty"`
 
-	//The ID of the vSwitch to which to connect the instance. This parameter is required when you create an instance of the VPC type. You can call the DescribeVSwitches operation to query the created vSwitches.
-	VSwitchID string `json:"vSwitchId,omitempty"`
-
 	//VSwitch is a reference to the vswitch to use for this instance
 	//This parameter is required when you create an instance of the VPC type.
 	//You can call the DescribeVSwitches operation to query the created vSwitches.
-	VSwitch *ResourceTagReference `json:"vSwitch,omitempty"`
+	VSwitch *AlibabaResourceReference `json:"vSwitch,omitempty"`
```

Comment:
Does this need to be a pointer?

Reply:
I saw examples in the AWS where it was so I went with that. I now also see examples where it does not need to be. I'll try and remove the pointer.